core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.component.ts

```diff
@@ -54,25 +51,18 @@ export class AIImagePromptComponent {
         this.store.hideDialog();
     }
 
-    /**
-     * Selects the prompt type
-     *
-     * @return {void}
-     */
-    selectType(promptType: PromptType, current: PromptType): void {
-        if (current != promptType) {
-            this.store.setPromptType(promptType);
-        }
-    }
-
     /**
      * Generates an image based on the provided prompt.
      *
-     * @param {string} prompt - The text prompt used to generate the image.
+     * @param {AIImagePrompt} formValue - The object prompt used to generate the image.
      * @return {void} - This method does not return any value.
      */
-    generateImage(prompt: string): void {
-        this.store.generateImage(prompt);
+    generateImage(formValue: AIImagePrompt): void {
+        this.store.generateImage(formValue);
+    }
+
+    insetImage(imageInfo: DotGeneratedAIImage): void {
+        this.store.patchState({ selectedImage: imageInfo });
```

Comment:
Yes, an updater. It is more easy to test too.

Reply:
done